### ISSUE TITLE
Pulling images from a private registry doesn't seem to work

### ISSUE THREAD

**Issue by ashoksekar07** [kubernetes-sigs/cluster-api-provider-aws#1222]
/kind bug

**What steps did you take and what happened:**
I created a Kubernetes cluster using CAPA. I tried to create a pod which uses an image from a private registry. The pod fails when pulling the image with 
`no available registry endpoint: unexpected status code https://<registry>/<image>: 401 Unauthorized`.

I have the image pull secret configured correctly. And the same pod and image pull secret works on a Kind cluster and a different cluster outside of CAPA.

I've also verified that the CAPA cluster has connectivity to the registry. I used curl with the auth token to pull the image metadata to verify this.

**What did you expect to happen:**
I expected the pod to be successfully created by pulling the image from the private registry.

**Environment:**

- Cluster-api-provider-aws version: 0.3.9
- Kubernetes version: (use `kubectl version`): 1.14.6
- OS (e.g. from `/etc/os-release`): Ubuntu 18.04.3 LTS
- Containerd version: 1.2.8

**Comment by ncdc** [comment]
@ashoksekar07 do you have step by step instructions for how to reproduce?

**Comment by ashoksekar07** [comment]
**Steps to repro**

Create an image pull secret pointing to a container registry which requires authentication. The image pull secret should be of type `dockerconfigjson` with the json content like this:
```
{
    "auths": {
        "https://<registry-endpoint>":{
            "username":"user",
            "password":"password",
            "email":"email@example.com"
        }
    }
}
```
Then create a pod which uses the image pull secret to pull its image:
```
apiVersion: v1
kind: Pod
metadata:
  name: demo
spec:
  containers:
  - name: demo
    image: <image>
  imagePullSecrets:
  - name: image-pull-secret
```
Pod should fail with an image pull error.

**Comment by ashish-amarnath** [comment]
@ashoksekar07 suggest doing a `base64 --decode` on the secret data to confirm that the credentials being used are correct.
Also, this is an upstream kubernetes behavior and CAPA doesn't do anything to alter it.

**Comment by ashoksekar07** [comment]
@ashish-amarnath I've verified the secret and I've used the same secret to successfully pull image on a Kind cluster and another cluster that was not setup using CAPA. I've also verified that from within my CAPA instance, I have connectivity to the registry.

So, my suspicion is that the AMI that comes with CAPA has some missing configuration or the containerd version that ships with this AMI has some bug.
